lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -181,7 +181,9 @@ SPIRVToLLVMDbgTran::transCompilationUnit(const SPIRVExtInst *DebugInst) {
 DIBasicType *SPIRVToLLVMDbgTran::transTypeBasic(const SPIRVExtInst *DebugInst) {
   using namespace SPIRVDebug::Operand::TypeBasic;
   const SPIRVWordVec &Ops = DebugInst->getArguments();
-  assert(Ops.size() == OperandCount && "Invalid number of operands");
+  assert((Ops.size() == OperandCountOCL ||
+          Ops.size() == OperandCountNonSemantic)
+         && "Invalid number of operands");
```

Comment:
Ok, even if we do nothing for these flags in forward translation (can we produce some default value to be spec conformant?), I believe we still should add handling of flags in reverse, especially knowing that `createBasicType` does have Flags argument. Does it make sense?

Reply:
Unfortunately - no. Default flag is no flag aka DINode::FlagZero (it's already default parameter of createBasicType in DIBuilder).

There is no 'unknown physical layout' base type entity in C, C++, Fortran and GLSL. There we can only have structures, that usually have hardware dependent physical layout (aka unknown for this compilation stage). Also there is a close thing in Rust called 'unsafe', but TBH I didn't check what Rust FE with -g would generate for it.

Also one create 'pseudo base type' with unknown layout in assembly (not our case) and may be in shaders, but I didn't try it (anyway it's not LLVM IR).

Reply:
Please fix clang-format here